modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/SkinDisposeContractTest.java

```diff
@@ -0,0 +1,201 @@
+import javafx.scene.control.TitledPane;
+import javafx.scene.control.ToggleButton;
+import javafx.scene.control.ToolBar;
+import javafx.scene.control.TreeCell;
+import javafx.scene.control.TreeTableCell;
+import javafx.scene.control.TreeTableRow;
+import javafx.scene.control.TreeTableView;
+import javafx.scene.control.TreeView;
+
+/**
+ * Test for https://bugs.openjdk.java.net/browse/JDK-8244112:
+ * skin must not blow if dispose is called more than once.
+ * <p>
+ * This test is parameterized in the type of control.
+ */
+@RunWith(Parameterized.class)
+public class SkinDisposeContractTest {
+
+    private Control control;
+    private Class<Control> controlClass;
+
+    /**
+     * Skin must support multiple calls to dispose.
+     * <p>
+     * default -> dispose -> dispose
+     * <p>
+     * Errors on second dispose are JDK-8243940.
+     * Failures/errors on first dispose (or before) are other errors - controls
+     * are commented with issue reference
+     *
+     */
+    @Test
+    public void testDefaultDispose() {
+        installDefaultSkin(control);
+        control.getSkin().dispose();
+        control.getSkin().dispose();
+    }
+
+  //---------------- parameterized
+
+    // Note: name property not supported before junit 4.11
+    @Parameterized.Parameters //(name = "{index}: {0} ")
+    public static Collection<Object[]> data() {
+        // class of control to test
+        // commented controls have different issues as described in the referenced issues
+        Object[][] data = new Object[][] {
+            {Accordion.class, },
+            {Button.class, },
+            {ButtonBar.class, },
+            {CheckBox.class, },
+            {ChoiceBox.class, },
+            {ColorPicker.class, },
+            {ComboBox.class, },
+            {DateCell.class, },
+            {DatePicker.class, },
+            {Hyperlink.class, },
+            {Label.class, },
+            {ListCell.class, },
+            {ListView.class, },
+            {MenuBar.class, },
+            {MenuButton.class, },
+            {Pagination.class, },
+            {PasswordField.class, },
+            {ProgressBar.class, },
+            {ProgressIndicator.class, },
+            {RadioButton.class, },
+            {ScrollBar.class, },
+            {ScrollPane.class, },
+            {Separator.class, },
+            {Slider.class, },
+            {Spinner.class, },
+            {SplitMenuButton.class, },
+            {SplitPane.class, },
+            {TableCell.class, },
+            {TableRow.class, },
+            {TableView.class, },
+            {TabPane.class, },
+            // @Ignore("8244419")
+            // {TextArea.class, },
+            {TextField.class, },
+            {TitledPane.class, },
+            {ToggleButton.class, },
+            {ToolBar.class, },
+            {TreeCell.class, },
+            {TreeTableCell.class, },
+            {TreeTableRow.class, },
+            {TreeTableView.class, },
+            {TreeView.class, },
```

Comment:
Should these controls be included too: `Tooltip`, `HTMLEditor`, `ContextMenu` ?

Reply:
good catch :) 

Hmm .. thinking aloud: 

- all in web is a bit very different as a small layer on top of the webEngine, usually off my radar
- the skinnable windows are no controls

so the easy way out - which I would tend take :) - is to narrow the issue to include only controls in controls. What do you think?

Reply:
Agree that HTMLEditorSkin is a web module. let's keep it off radar :)

TooltipSkin and ContextMenuSkin implement Skin and override the dispose() method. I think these Skins should also behave similar to other and can be included here in test. and also should consider them under cleanup issue [JDK-8241364](https://bugs.openjdk.java.net/browse/JDK-8241364).
Both the dispose() seem safe from NPE.

Reply:
okay, added a separate test for ContextMenu/Tooltip (they don't fit into the binch test here :)